test/mip_test.c

```diff
@@ -49,9 +67,108 @@ static void test_statechange(void) {
   onstatechange(&iface);
 }
 
+// *****************************************************************************
+// Implement tap_rx() tap_tx() tap_up()
+
+static size_t tap_rx(void *buf, size_t len, void *userdata) {
+  ssize_t received = read((int) userdata, buf, len); // Original
```

Comment:
As discussed, please try this to squash the warning:

```c
read((int) (size_t) userdata, buf, len);
```

If that does not work, search for other ways on how to do a cast without a warning.